packages/wallet/api/src/marshal-contexts.js

```diff
@@ -0,0 +1,151 @@
+// @ts-check
+import { makeScalarMap } from '@agoric/store';
+import { Far, makeMarshal } from '@endo/marshal';
+
+const { details: X, quote: q } = assert;
+
+const makePresence = (_slot, iface) => {
+  const severed = `SEVERED: ${iface.replace(/^Alleged: /, '')}`;
+  const thing = /** @type {any} */ (Far(severed, {}));
+  return thing;
+};
+
+/**
+ * Make context for exporting wallet data where brands etc. can be recognized by boardId.
+ *
+ * All purses, brands, etc. must be registered before serialization / unserialization.
+ * Neither the serializer nor the unserializer creates new presences.
+ */
+export const makeExportContext = () => {
+  const toVal = {
+    /** @type {MapStore<string, Purse>} */
+    purse: makeScalarMap(),
+  };
+  const fromVal = {
+    /** @type {MapStore<Purse, string>} */
+    purse: makeScalarMap(),
+  };
+  /** @type {MapStore<unknown, string>} */
+  const sharedData = makeScalarMap();
+
+  /**
+   * @param {string} slot
+   * @param {string} _iface
+   */
+  const slotToVal = (slot, _iface) => {
+    const kind = Object.keys(toVal).find(k => slot.startsWith(k));
```

Comment:
no need to bind a delegating function
```suggestion
    const kind = Object.keys(toVal).find(slot.startsWith);
```

Reply:
I don't think that works.  `startsWith` expects `this` to be bound to the receiver (`slot`):

```js
> ["purse", "brand"].find("brand:1".startsWith)
VM400:1 Uncaught TypeError: String.prototype.startsWith called on null or undefined
```

It works with our objects-as-closures where we pretend `this` isn't part of the language :)

Reply:
ah, thanks for explaining